velociraptor/catalogue/soap_catalogue.py

```diff
@@ -12,13 +11,63 @@
 
 from functools import reduce
 
-from astropy.cosmology import wCDM, FlatLambdaCDM
 import unyt
+from swiftsimio.conversions import swift_cosmology_to_astropy
+from swiftsimio.metadata.unit.unit_types import find_nearest_base_unit
+from swiftsimio.metadata.unit.unit_types import unit_names_to_unyt
 
 from abc import ABC, abstractmethod
 from pathlib import Path
 
 
+class SWIFTUnits_from_dictionary(object):
```

Comment:
I think this name is really weird, on a second look. The mixing of camel and snake case is really odd, and it's hard to tell if this is a function or a class.

How about `SWIFTUnitsMockup`?